src/main/java/rip/bolt/ingame/ranked/RankedManager.java

```diff
@@ -109,17 +112,43 @@ private void updateMatch(BoltMatch match) {
       return;
     }
 
-    List<MatchPlayer> players =
+    MatchManager mm = PGM.get().getMatchManager();
+
+    List<UpdateRecord> updates =
         match.getTeams().stream()
             .map(Team::getParticipations)
             .flatMap(Collection::stream)
             .map(Participation::getUser)
-            .map(rankManager::notifyUpdates)
-            .filter(Objects::nonNull)
+            .map(
+                u ->
+                    new UpdateRecord(u, this.match.getUser(u.getUuid()), mm.getPlayer(u.getUuid())))
+            .filter(UpdateRecord::isValid)
             .collect(Collectors.toList());
 
     this.match = match;
-    players.forEach(rankManager::updatePlayer);
+    if (updates.isEmpty()) return;
```

Comment:
there isn't a case when this can be empty right? as players wouldn't get to see their stats if it returned here. assume this is to satisfy the `updates.get(0)` a bit further down

Reply:
If all players in the teams leave this will be empty